README.md

```diff
@@ -1,2 +1,72 @@
-# faraday_connection_pool
+# FaradayConnectionPool
+
 FaradayConnectionPool provides a persistent Net::HTTP Faraday adapter.
+Unlike Net::Persistent, connections are pooled across threads and fibres so that they can be reused more frequently.
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+    gem 'faraday_connection_pool'
+
+And then execute:
+
+    $ bundle
+
+Or install it yourself as:
+
+    $ gem install faraday_connection_pool
+
+
+## Usage
+
+1. If necessary `require 'faraday_connection_pool'`
+
+2. Configure FaradayConnectionPool:
+
+```ruby
+FaradayConnectionPool.configure do |config|
+  config.size = 5 #The number of connections to held in the pool. There is a separate pool for each host/port.
+  config.timeout = 0.5 #If no connection is available from the pool within :timeout seconds the adapter will raise a Timeout::Error.
+end
+```
+
+3. Configure your Faraday connections to use the `:net_http_pooled` adapter provided by the gem:
+
+```ruby
+Faraday.new(:url => 'http://klarna.com') do |conn|
+  conn.adapter :net_http_pooled
+end
+```
+
+## Warning - Retries
+
+FaradayConnectionPool will not automatically try and repair broken connections, so you should configure Faraday to retry
+for you:
+
+```ruby
+Faraday.new do |conn|
+  conn.request :retry, max: 2, interval: 0.05,
+                       interval_randomness: 0.5, backoff_factor: 2
+                       exceptions: [CustomException, 'Timeout::Error']
```

Comment:
I wish it was split out more, honestly... but, since it's not and you're only documenting how to use it, then you could include that in the retry logic and maybe expand a bit on _why_ you did that.